torchvision/prototype/datasets/_builtin/cub200.py

```diff
@@ -173,9 +177,8 @@ def _make_datapipe(
             )
 
             image_files_dp = CSVParser(image_files_dp, dialect="cub200")
-            image_files_map = dict(
-                (image_id, rel_posix_path.rsplit("/", maxsplit=1)[1]) for image_id, rel_posix_path in image_files_dp
-            )
+            image_files_dp = Mapper(image_files_dp, self._2011_image_key, input_col=1)
```

Comment:
What we can do here is a `in_memory_cache` over `image_files_dp`. Then, we could add an API to convert a `IterDataPipe` to a lazy loaded `MapDataPipe` to represent the `LazyDict`.
If we use `LazyDict` here, I have concern that `image_files_map` would be missing from the DataLoader graph.
cc: @VitalyFedyunin

Reply:
Besides, I think the DataLoader would complain this datapipe graph in the second epoch because `image_files_dp` is never used after the first epoch then `Demux` would also be non-serializable same as the comment I made in the other PR.

So, a fix from `Demux` is not avoidable.

Reply:
This seems like a good thing to test in general. What should a test look like. Is something like

```python
for _ in dataset.cycle(2):
    pass
```

enough? If yes, my proposal passes this test.

Reply:
I mean if we put the dataset (datapipes) into DataLoader, the second epoch of DataLoader would break.

Reply:
So something like

```python
data_loader = DataLoader2(dataset)

for epoch in range(2):
    for sample in data_loader:
        pass
```

?

Reply:
Yeah.

Reply:
My proposal still works. I've pushed the test I'm running against. There are multiple failures for other datasets, but `cub200` is not one of them.